aspnetcore/grpc/troubleshoot/sample/8.0/GrpcGreeterClient/Program.cs

```diff
@@ -0,0 +1,204 @@
+﻿#define StandardHTTPS  // Options: StandardHTTPS | IgnoreInvalidCertificate | IgnoreInvalidCertificateClientFactory | CallInsecureGrpcServices | DotNet3InsecureGrpcServices | SubdirectoryHandler | Http3Handler
+
+#if StandardHTTPS
+using Grpc.Net.Client;
+using GrpcGreeterClient;
+
+// The port number must match the port of the gRPC server.
+// <snippet_StandardHTTPS>
+var channel = GrpcChannel.ForAddress("https://localhost:5001");
+var client = new Greeter.GreeterClient(channel);
+// </snippet_StandardHTTPS>
+var reply = await client.SayHelloAsync(
+                  new HelloRequest { Name = "GreeterClient" });
+Console.WriteLine("Greeting: " + reply.Message);
+Console.WriteLine("Press any key to exit...");
+Console.ReadKey();
+#endif
+
+#if IgnoreInvalidCertificate
+// Warning: Untrusted certificates should only be used during app development. 
+// Production apps should always use valid certificates.
+// The following gRPC client factory allows calls without a trusted certificate.
+
+using System.Threading.Tasks;
+using Grpc.Net.Client;
+using GrpcGreeterClient;
+
+// <snippet_IgnoreInvalidCertificate>
+var handler = new HttpClientHandler();
+handler.ServerCertificateCustomValidationCallback = 
+    HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+
+var channel = GrpcChannel.ForAddress("https://localhost:5001",
+    new GrpcChannelOptions { HttpHandler = handler });
+var client = new Greeter.GreeterClient(channel);
+// </snippet_IgnoreInvalidCertificate>
+
+var reply = await client.SayHelloAsync(
+                  new HelloRequest { Name = "GreeterClient" });
+Console.WriteLine("Greeting: " + reply.Message);
+Console.WriteLine("Press any key to exit...");
+Console.ReadKey();
+#endif
+
+#if IgnoreInvalidCertificateClientFactory
+// Warning: Untrusted certificates should only be used during app development. 
+// Production apps should always use valid certificates.
+// The following gRPC client factory allows calls without a trusted certificate.
+
+using GrpcGreeterClient;
+using Microsoft.Extensions.DependencyInjection;
+using Grpc.Net.ClientFactory;
+
+// <snippet_IgnoreInvalidCertificateClientFactory>
+
```

Comment:
@JamesNK, I am using ServiceCollection() here in the example for using a client factory with a console client.  It seemed to work right for the console app example, but it is different than the example you had, which seemed like it was geared for a web client.  I don't have any of the original app examples used for this, so I wasn't sure.  So I'll point out this difference so it is not missed just in case.